test/functions.js

```diff
@@ -694,6 +694,9 @@
       assert.deepEqual(_.toArray(cb(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)), _.range(1, 11));
     });
 
+    var deepProperty = _.iteratee(['a', 'b']);
+    assert.equal(deepProperty({a: {b: 2}}), 2, 'treats an array as a deep property accessor');
+
```

Comment:
Assuming Underscore is philosophically opposed to the `a.b.c` syntax, I think it makes most sense to treat arrays as deep property matching:

    // Underscore only (proposed)
    var adminPosts = _.filter(posts, ['author', 'isAdmin']);

    // Lodash only
    var adminPosts = _.filter(posts, 'author.isAdmin');
    var adminPosts = _.filter(posts, {author, {isAdmin: true}});

    // Compat (proposed)
    var adminPosts = _.filter(posts, _.property(['author', 'isAdmin']));